### DISCUSSION TITLE
Svelte: Reinitializing client and invalidating cache auth bearer token changes

### DISCUSSION THREAD

**Discussion by johnmerchant** [urql-graphql/urql#1727] [Q&A] [answered]
Howdy Urql'ers 👋🏼 

With `@urql/svelte` and the following Svelte init `<script>`, what would be the best way to invalidate the query cache and refetch all queries when the `token` prop changes? 

For context, this is for a Single-Page App that can login and receive a token to authenticate with an API.

```typescript
import { initClient } from "@urql/svelte";
export let token: string | undefined;
initClient({
   url: 'https://example.com/graphql',
   fetchOptions: {
       headers: token ? { Authorization: `Bearer ${token}` } : {}
   } 
});
```

**Comment by kitten** [comment, marked as answer]
I haven't tried this yet per se, since I don't actively use Svelte that often, but isn't it possible to create a reactive block for this?

```js
import { initClient } from "@urql/svelte";
export let token;

$: {
  initClient({
     url: 'https://example.com/graphql',
     fetchOptions: {
         headers: token ? { Authorization: `Bearer ${token}` } : {}
     } 
  });
}
```

**Reply by johnmerchant** [reply]
Using a reactive block does update the `fetchOptions`, but it does not trigger a refetch of queries declared in other components. Another solution I found is to use the fetchOptions callback:

```javascript
fetchOptions() {
    return { headers: token ? { Authorization: `Bearer ${token}` } : {} };
}
```

Perhaps the issue is with how my components are designed, rather than needing to trigger a refetch of all queries. Conditionally rendering components that use authenticated queries in `{#if isAuthenticated}` blocks works well.
